contracts/lupi.sol

```diff
@@ -15,17 +20,35 @@ contract lupi is owned {
     State state = State.Betting;
 
     Ticket[] tickets;
-    uint revealedCount;
+    uint revealedCount; // TODO: shall we replace this with a getter returning revealedBets.length?
 
     // bet => tickets
     mapping(uint => uint[]) revealedTickets;
     uint[] revealedBets;
 
     uint winningTicket;
 
-    function lupi() {
+    // TODO: function getBets(address _player) constant returns bets[]
+
+    function lupi(uint _requiredBetAmount, uint _ticketCountLimit, uint _feePt ) {
+        requiredBetAmount = _requiredBetAmount;
+        ticketCountLimit = _ticketCountLimit;
+        feePt = _feePt;
         // ticket zero is reserved
-        tickets.push(Ticket(0, 0, 0, 0));
+        tickets.push(Ticket(0, 0, 0, 0)); // TODO: shall we replace this for a more readable logic?
+    }
+
+    function getRoundInfo() constant returns ( // FIXME: state doesn't seem to work
+            State _state, uint _requiredBetAmount, uint _feePt, uint _ticketCountLimit,
+            uint _ticketCount, uint _revealedCount,
+            uint  winnablePotAmount) {
+        return ( state, requiredBetAmount, feePt, ticketCountLimit,
+            tickets.length -1, revealedBets.length,
+            (tickets.length -1) * requiredBetAmount - getFeeAmount());
+    }
+
+    function getFeeAmount() constant returns (uint feeAmount) {
```

Comment:
we need to calc fee at multiple places, having a getter makes the code more readable